rosidl_typesupport_introspection_c/resource/msg__type_support.c.em

```diff
@@ -198,9 +199,15 @@ static rosidl_message_type_support_t @(function_prefix)__@(spec.base_type.type)_
   &@(function_prefix)__@(spec.base_type.type)_message_members
 };
 
-ROSIDL_TYPESUPPORT_INTROSPECTION_C_EXPORT_@(spec.base_type.pkg_name)
+ROSIDL_GENERATOR_C_EXPORT_@(spec.base_type.pkg_name)
```

Comment:
So long as it's `*_EXPORT*`, you can use any of them (including the one @dirk-thomas is suggesting). `*_EXPORT*` always evaluates to `__declspec(dllexport)` on Windows, regardless of what `-D` options are passed to the compiler.

It is still true, however, that the declaration of this function uses the `ROSIDL_GENERATOR_C_PUBLIC_*` macro and at least in my opinion, it would make sense if it matched, since this is just the definition of that function declared in the `rosidl_generator_c` code and not a new function that is specific to `rosidl_typesupport_introspection_c`.